plugins/database/hana/hana.go

```diff
@@ -148,148 +152,222 @@ func (h *HANA) CreateUser(ctx context.Context, statements dbplugin.Statements, u
 				"password":   password,
 				"expiration": expirationStr,
 			}
+
 			if err := dbtxn.ExecuteTxQuery(ctx, tx, m, query); err != nil {
-				return "", "", err
+				return newdbplugin.NewUserResponse{}, err
 			}
 		}
 	}
 
 	// Commit the transaction
 	if err := tx.Commit(); err != nil {
-		return "", "", err
+		return newdbplugin.NewUserResponse{}, err
+	}
+
+	resp := newdbplugin.NewUserResponse{
+		Username: username,
 	}
 
-	return username, password, nil
+	return resp, nil
 }
 
 // Renewing hana user just means altering user's valid until property
-func (h *HANA) RenewUser(ctx context.Context, statements dbplugin.Statements, username string, expiration time.Time) error {
-	statements = dbutil.StatementCompatibilityHelper(statements)
+func (h *HANA) UpdateUser(ctx context.Context, req newdbplugin.UpdateUserRequest) (newdbplugin.UpdateUserResponse, error) {
+	h.Lock()
```

Comment:
Asking because I have the same question for the Redshift DB migration: why do we need to hold a mutex lock inside the plugin? I can see that Vault needs to serialise password updates so that it knows which is the current password after concurrent rotate requests. However, the state is stored outside of this plugin, so the synchronisation would also need to be outside of the plugin i.e. around the state update in the caller. Am I missing another synchronisation requirement? cc @pcman312

Reply:
This was protecting against races when multiple calls are made to the engine at the same time. Specifically: rotate root credentials + any other call since rotate root was changing the plugin instance's configuration. I'm not 100% sure this is needed any longer, but I'd like to keep it for safety until we can confirm that it isn't needed.